src/app/components/ProductDetails/ProductDetails.js

```diff
@@ -1,16 +1,62 @@
-import React from 'react';
+import React, { Component } from 'react';
 import ProductDescription from '../ProductDescription/ProductDescription';
+import RelatedProducts from '../RelatedProducts/RelatedProducts';
+import HttpService from '../../../utils/http.service';
 
-const ProductDetails = props => {
-  const { id, products } = props;
+class ProductDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { relatedProducts: [] };
+  }
 
-  const item = products.find(el => el._id === id);
+  componentDidMount() {
+    this.forceUpdate();
+    window.scrollTo(0, 0);
+  }
 
-  return (
-    <div className="container">
-      <ProductDescription item={item} />
-    </div>
-  );
-};
+  componentDidUpdate() {
+    const { products, id } = this.props;
+    const { relatedProducts } = this.state;
+    const item = products.find(el => el._id === id);
+
+    if (item && !relatedProducts.length) {
+      const params = {
```

Comment:
would be better to pass only needed filter params

and override default params

I suggest using some service class to handle this